### ISSUE TITLE
Parsing encoding problem with .html() on <style> tag

### ISSUE THREAD

**Issue by mino181295** [jhy/jsoup#1186]
I have a problem parsing and then setting the _<style>_ tag.
When I do .html() and then set back the string with .html(s) I get the HTML entity &lt; that does not work correctly in CSS.

When I do:
```
        String html = ""
                + "<html>"
                + " <head>"
                + "     <style>.outer > .inner {background-color:white;}</style>"
                + " </head>"
                + " <body>"
                + "     Example"
                + " </body>"
                + "</html>";
        Document document = Jsoup.parse(html);
        Element style = document.selectFirst("style");
        
        String s = style.html();
        //..
        style.html(s);
        
        System.out.println(document.toString());
```
I get:

```
<html>
 <head>
  <style>.outer &gt; .inner {background-color:white;}</style>
 </head>
 <body>
   Example
 </body>
</html>
```

But I'm expecting that the style tag preserve the **>** selector or it won't work correctly.
I'm expecting that the result HTML code is:
```
<html>
 <head>
  <style>.outer > .inner {background-color:white;}</style>
 </head>
 <body>
   Example
 </body>
</html>
```

I would be happy to contribute to an eventual patch if you give me some hints.

**Comment by mino181295** [comment]
@jhy

**Comment by lnostdal** [comment]
Work on a clone in order to avoid mutating the `document` instance. Essentially the same as your code, but in Clojure

```clj
symbolicweb.core> (let [document (Jsoup/parse (str ""
                                         "<html>"
                                         " <head>"
                                         "     <style>.outer > .inner {background-color:white;}</style>"
                                         " </head>"
                                         " <body>"
                                         "     Example"
                                         " </body>"
                                         "</html>"))
                        style (.selectFirst document "style")]
                    (println (.toString (.html style (.html style))))
                    (println "\n###\n")
                    (println (.toString document)))
<style>.outer &gt; .inner {background-color:white;}</style>

###

<html>
 <head> 
  <style>.outer &gt; .inner {background-color:white;}</style> 
 </head> 
 <body>
   Example 
 </body>
</html>
nil
symbolicweb.core> 
```

...when we use .clone, it is fixed:

```clj
symbolicweb.core> (let [document (Jsoup/parse (str ""
                                         "<html>"
                                         " <head>"
                                         "     <style>.outer > .inner {background-color:white;}</style>"
                                         " </head>"
                                         " <body>"
                                         "     Example"
                                         " </body>"
                                         "</html>"))
                        style (.clone (.selectFirst document "style"))]
                    (println (.toString (.html style (.html style))))
                    (println "\n###\n")
                    (println (.toString document)))
<style>.outer &gt; .inner {background-color:white;}</style>

###

<html>
 <head> 
  <style>.outer > .inner {background-color:white;}</style> 
 </head> 
 <body>
   Example 
 </body>
</html>
nil
symbolicweb.core> 
```

**Comment by eolivelli** [comment]
I have found the problem.
In @mino181295 case we have to mutate the STYLE element (the example was not complete).

I am sending a Pull Request with an example in form of test case
#1186

**Comment by mino181295** [comment]
Thank you @eolivelli that solved the problem.

**Comment by jhy** [comment]
This was fixed with #1419
